src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -164,3 +186,90 @@ def create(self, stage_name: str, comment: Optional[str] = None) -> SnowflakeCur
     def iter_stage(self, stage_path: str):
         for file in self.list_files(stage_path).fetchall():
             yield file["name"]
+
+    def execute(
+        self,
+        stage_path: str,
+        on_error: OnErrorType,
+        variables: Optional[List[str]] = None,
+    ):
+        all_files_list = self._get_files_list_from_stage(stage_path)
+        # filter files from stage if match stage_path pattern
+        filtered_file_list = self._filter_files_list(stage_path, all_files_list)
+
+        if not filtered_file_list:
+            raise ClickException(f"No files matched pattern '{stage_path}'")
+
+        # sort filtered files in alphabetical order with directories at the end
+        sorted_file_list = sorted(
+            filtered_file_list, key=lambda f: (path.dirname(f), path.basename(f))
+        )
+
+        sql_variables = self._parse_execute_variables(variables)
+        results = []
+        for file in sorted_file_list:
+            results.append(self._call_execute_immediate(file, sql_variables, on_error))
+
+        return results
+
+    def _get_files_list_from_stage(self, stage_path: str) -> List[str]:
+        stage_name = self.get_stage_name_from_path(stage_path)
+        files_list_result = self.list_files(stage_name).fetchall()
+
+        if len(files_list_result) == 0:
+            raise ClickException(f"No files found on stage '{stage_name}'")
+
+        return [f["name"] for f in files_list_result]
+
+    def _filter_files_list(
+        self, stage_path: str, files_on_stage: List[str]
+    ) -> List[str]:
+        stage_path = self.remove_stage_prefix(stage_path)
+
+        # Execute everything if only stage name was provided
+        if "/" not in stage_path:
+            return files_on_stage
+
+        # Filter with fnmatch if contains `*` or `?`
+        if glob.has_magic(stage_path):
+            return fnmatch.filter(files_on_stage, stage_path)
+
+        # Exact file path was provided if stage_path in file list
+        if stage_path in files_on_stage:
+            return [stage_path]
+
+        # Path to directory was provided
+        return fnmatch.filter(files_on_stage, f"{stage_path}*")
+
+    @staticmethod
+    def _parse_execute_variables(variables: Optional[List[str]]) -> Optional[str]:
+        if not variables:
+            return None
+
+        query_parameters = []
+        for p in variables:
+            if "=" not in p:
+                raise ClickException(f"Invalid variable: '{p}'")
+
+            key, value = p.split("=")
```

Comment:
What about `"key=val=ue""` ?

Reply:
Good point, check and test added

Reply:
Why is it invalid? I think it's valid, we just have to make the parsing right. `p.split("=", 1)` should solve the issue.

I would consider extracting this parsing logic to separate method that would return a list of {key: ..., value: ...}  dictionaries. There will be other places where we will want to use similar parsing.

Reply:
Extracted